src/os/bluestore/BlueStore.cc

```diff
@@ -3650,6 +3656,10 @@ void BlueStore::ExtentMap::reshard(
   dout(20) << __func__ << "  extent_avg " << extent_avg << ", target " << target
 	   << ", slop " << slop << dendl;
 
+  uint32_t next_boundary = segment_size;
+  uint32_t encoded_segment_estimate = bytes * segment_size / (data_reshard_end - needs_reshard_begin);
```

Comment:
shouldn't we have an additional protection (better something less destructive than assert) against data_reshard_end == needs_reshard_begin?

Reply:
But it always will be `needs_reshard_begin < data_reshard_end`.
Assert would be exactly my choice in this situation, if anything.

Reply:
IMO there is no clear straightforward logic to prove that statement.
And assertion is not much way better than hitting division by zero exception. Both aren't bypassable in the field and effectively kill an OSD which faces that..

Reply:
Inserted code to avoid div/0.
If it ever happens, derr is also triggered.